mantidproject/mantid
# Switch to JSON Parser for SANS State

**Description of work.**
Switches to using the JSON library for parsing SANS state objects. This
provides numerous advantages:

- We do not need to maintain a custom (de)serializer
- JSON is a documented standard
- We can switch to type hinting in Python 3, currently a significant
amount of CPU time is spend reverifying typed params
- History becomes less brittle and will work from top level algorithms
- Allows new code to be written in a more Pythonic way (e.g. not forced
to use class level variables for them to be serialized)


The time justification is two fold:
- New types needed to be added as part of TOML support, adding to the maintenance burden of a new parser
- In Python 3.9 named tuples (in collections) will be deprecated. This will require maintenance on the parser (again)

So it made sense to clear it down now

**To test:**
- Open the SANS GUI and process some data and check the results are similar
- Check all tests pass
- Code review

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
